client/state/user-settings/reducer.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+import { merge, omit } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import {
+	USER_SETTINGS_FETCH,
+	USER_SETTINGS_FETCH_SUCCESS,
+	USER_SETTINGS_FETCH_FAILURE,
+	USER_SETTINGS_SAVE_SUCCESS,
+	USER_SETTINGS_RECEIVE,
+	USER_SETTINGS_UNSAVED_SET,
+	USER_SETTINGS_UNSAVED_REMOVE,
+} from 'state/action-types';
+import { createReducer } from 'state/utils';
+import { decodeEntities } from 'lib/formatting';
+
+/*
+ * Decodes entities in those specific user settings properties
+ * that the REST API returns already HTML-encoded
+ */
+function decodeUserSettingsEntities( data ) {
+	const decodedValues = {
+		display_name: data.display_name && decodeEntities( data.display_name ),
+		description: data.description && decodeEntities( data.description ),
+		user_URL: data.user_URL && decodeEntities( data.user_URL )
+	};
```

Comment:
awesome job normalizing the data from the API!

unfortunately, remember that this _isn't_ technically data coming in from the API…what's that? It's actually data coming in from the dispatcher, which in most cases happens to come originally from the API.

it would be wonderful if the encoded information never ever entered the Calypso world by normalizing where we get the network request. this of course has its spot in the data layer files where the entire mapping stays in one easily-discoverable spot.

for example, if I want to test some code with a given `display_name` then I'd just as soon open the dev tools and dispatch an action with a predefined name but I don't want to have to know that I need to first encode that string because the API does…